R/layer.r

```diff
@@ -299,6 +299,9 @@ Layer <- ggproto("Layer", NULL,
     # evaluation (since the evaluation symbols gets renamed)
     data <- rename_aes(data)
 
+    # data needs to be non-scaled
+    data_orig <- scales_backtransform_df(plot$scales, data)
```

Comment:
I think you might be able to alleviate Thomas' concern that this is applied to 99% of plots, by executing this line after the early exit at the `if (length(new) == 0) return(data)` line.

Reply:
Agreed, thanks!